File: backend/app/schemas/checkin.py
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class CheckinCreate(BaseModel):
    energy_level: int = Field(..., ge=1, le=5)
    soreness: int = Field(..., ge=1, le=5)
    perceived_sleep_hours: float = Field(..., ge=0, le=24)
    general_feeling: int = Field(..., ge=1, le=5)


class CheckinResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    date: date
    energy_level: int
    soreness: int
    perceived_sleep_hours: float
    general_feeling: int
    workout_generated: Optional[dict] = None
    agent_message: Optional[str] = None
    created_at: datetime

File: mobile/src/types/checkin.ts
export interface CheckinCreate {
  energy_level: number      // 1-5
  soreness: number          // 1-5
  perceived_sleep_hours: number  // 0-12
  general_feeling: number   // 1-5
}

export interface AgentInsight {
  resumen: string
  relacion_sueno_readiness: string
  aporte_entrenamientos: string
  tendencia_7_dias: string
  sugerencia_dia: 'fuerte' | 'moderar' | 'descansar'
  sugerencia_detalle: string
  tono: 'positivo' | 'neutro' | 'precaucion'
}

export interface CheckinResponse {
  id: number
  date: string
  energy_level: number
  soreness: number
  perceived_sleep_hours: number
  general_feeling: number
  workout_generated: AgentInsight | null
  agent_message: string | null
  created_at: string
}
